src/librustdoc/clean/types.rs

```diff
@@ -1301,19 +1301,19 @@ crate struct Arguments {
     crate values: Vec<Argument>,
 }
 
-#[derive(Clone, PartialEq, Eq, Debug, Hash)]
-crate struct Argument {
-    crate type_: Type,
-    crate name: Symbol,
-}
-
 #[derive(Clone, PartialEq, Debug)]
 crate enum SelfTy {
     SelfValue,
     SelfBorrowed(Option<Lifetime>, Mutability),
     SelfExplicit(Type),
 }
 
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+crate struct Argument {
+    crate type_: Type,
+    crate name: Symbol,
+}
+
```

Comment:
Can you leave this part as it was before? Both `Argument` and `SelfTy` are logically connected to the following impl, so this feels like unnecessary churn. I'm likely going to be removing `SelfTy` soon anyway.

Reply:
Oh nice!